docker/docker_transport_test.go

```diff
@@ -121,6 +147,18 @@ func TestReferenceStringWithinTransport(t *testing.T) {
 		stringRef2 := ref2.StringWithinTransport()
 		assert.Equal(t, stringRef, stringRef2, c.input)
 	}
+
+	for _, c := range ValidReferenceUnknownDigestTestCases {
```

Comment:
Isn’t this exactly the same code as for `validReferenceTestCases`? If so, there could just be one loop over `append(slices.Clone(validReferenceTestCases), validReferenceUnknownDigestTestCases…)`